walk-in-interview/src/main/webapp/new-job.js

```diff
@@ -0,0 +1,74 @@
+
+const NEW_JOB_FORM_ID = 'new-job-form';
+const NEW_JOB_TITLE_ID = 'new-job-title';
+const NEW_JOB_ADDRESS_ID = 'new-job-address';
+const NEW_JOB_DESCRIPTION_ID = 'new-job-description';
+const NEW_JOB_PAY = {
+  FREQUENCY_ID: 'new-job-pay-frequency',
+  MIN_ID: 'new-job-pay-min',
+  MAX_ID: 'new-job-pay-max',
+};
+const REQUIREMENTS_LIST_ID = 'requirements-list';
+const NEW_JOB_EXPIRY_ID = 'new-job-expiry';
+const NEW_JOB_DURATION_ID = 'new-job-duration';
+
+/**
+ * Adds a new job listing by making a POST request to the /jobs servlet.
+ * @return {boolean} If the request given the parameters was valid or not.
+ */
+function addJob() {
```

Comment:
Optional, you don't really need a form here you can attach an event listener to the submit button and then perform the same operations as in the end the request is made async using AJAX.

Is there a specific reason to use `<form>` here?

Reply:
If you use the event listener approach then your `addJob` can return a `Promise` as well and you can consume it like

```js
function addJob() {} // returns a Promise

const submitButton = document.getElementById('submit_button_id');
submitButton. addEventListener('click', _ => {
    addJob()
    .then(ignore -> { 
        // on success things
    })
    .catch(failure -> {
        // failure cases
    });
});
```

Reply:
One more suggestion you can define Job as a class and `ECMA5` is supported in all major browsers now so something like

```js
class Job {
    constructor(jobInput) {}

    // async functions are similar to Promise, alternatively use submit() which returns a Promise
    async submitAsync() {}
    
    // factory method which could abstract getting data from DOM and creating job object
    static createFromUserInputs() {
        // get everything from DOM
        return new Job(jobInput);
    } 
}
```